### ISSUE TITLE
Array of ValueObjects diff issue

### ISSUE THREAD

**Issue by floresek** [javers/javers#207]
Hi again.

I'd like to report an diff isue.

```java
class Master {
    @Id
    String id;

    Detail[] array;

    public Master(String id) {
        this.id = id;
    }

}

@ValueObject
class Detail {
    String data;

    public Detail(String data) {
        this.data = data;
    }
}

        ...
		Master master = new Master("3");
        master.array = new Detail[] { new Detail("details-array")};
        javers.commit("anonymous", master);
        javers.commit("anonymous", master);
        javers.commit("anonymous", master);
        List<Change> changes = javers
                .findChanges(QueryBuilder.byInstanceId("3", Master.class)
                        .withNewObjectChanges(true)
                        .build());
        String changeLog = javers.processChangeList(changes, getChangeLogger());
		...
```
Second and third commit save Master snapshot with 'array' property changed (although no change occurred).
processChangeList traces only one initial change.

**Comment by bartoszwalacik** [comment]
if you are using sql repository, this issue could be fixed in fresh 1.3.10, could you check this?

Bug you've posted - https://github.com/javers/javers/issues/206 - was affecting snapshots comparision

**Comment by floresek** [comment]
The problem still exists in new version.

I ran the test with 1.3.10 javers version and java: jdk1.7.0_79 (64 bit)

Below I attached exported rows from jv_snapshot, which were created during commits:

```json
[{
  "snapshot_pk": 1400,
  "type": "INITIAL",
  "state": "{  "data": "details-array"}",
  "changed_properties": "[  "data"]",
  "global_id_fk": 1201,
  "commit_fk": 1500
 },
 {
  "snapshot_pk": 1401,
  "type": "INITIAL",
  "state": "{  "id": "3",  "array": [    {      "valueObject": "javers.Detail",      "ownerId": {        "entity": "javers.Master",        "cdoId": "3"      },      "fragment": "array/0"    }  ]}",
  "changed_properties": "[  "id",  "array"]",
  "global_id_fk": 1200,
  "commit_fk": 1500
 },
 {
  "snapshot_pk": 1402,
  "type": "UPDATE",
  "state": "{  "id": "3",  "array": [    {      "valueObject": "javers.Detail",      "ownerId": {        "entity": "javers.Master",        "cdoId": "3"      },      "fragment": "array/0"    }  ]}",
  "changed_properties": "[  "array"]",
  "global_id_fk": 1200,
  "commit_fk": 1501
 },
 {
  "snapshot_pk": 1403,
  "type": "UPDATE",
  "state": "{  "id": "3",  "array": [    {      "valueObject": "javers.Detail",      "ownerId": {        "entity": "javers.Master",        "cdoId": "3"      },      "fragment": "array/0"    }  ]}",
  "changed_properties": "[  "array"]",
  "global_id_fk": 1200,
  "commit_fk": 1502
 }
]
```

**Comment by bartoszwalacik** [comment]
okay, will take a look at this soon

**Comment by bartoszwalacik** [comment]
fix is on its way to Maven Central ...

**Comment by bartoszwalacik** [comment]
fixed in 1.3.13 release
